### PR TITLE
Port and run runtime tests

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
@pbchekin https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468952/job/20356718182 didn't catch the yapf failure caught by https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468953.

**Comment by pbchekin** [comment]
> @pbchekin https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468952/job/20356718182 didn't catch the yapf failure caught by https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468953.

I don't see anything reported by yapf in https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468953. Looks like yapf has crashed rather than worked normally and reported any changed files. A potential root cause for the yapf crash is race condition, see https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/.github/workflows/build_and_test_2.yaml#L51.

**Comment by whitneywhtsang** [comment]
> I don't see anything reported by yapf in https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468953. Looks like yapf has crashed rather than worked normally and reported any changed files. A potential root cause for the yapf crash is race condition, see https://github.com/intel/intel-xpu-backend-for-triton/blob/llvm-target/.github/workflows/build_and_test_2.yaml#L51.

```
yapf.....................................................................Failed
- hook id: yapf
- duration: 5.92s
- files were modified by this hook
```

And this is the file modification: 
```
--- a/python/test/unit/runtime/test_autotuner.py
+++ b/python/test/unit/runtime/test_autotuner.py
@@ -7,6 +7,7 @@ import pytest
 # FIXME remove this once Triton L0 queue and IPEX SYCL queue can be synchronized through events
 torch.xpu.enable_sync_mode()
 
+
 def test_kwargs():
     N = 1024
     src = torch.empty(N, device='xpu')
```

**Comment by pbchekin** [comment]
> And this is the file modification:

Strange, but I do not see this modification in https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/7479468953/job/20356717890. Anyway, good catch, should be fixed in https://github.com/intel/intel-xpu-backend-for-triton/pull/230.

**Review by ESI-SYD — APPROVED** [review]

File: python/test/unit/runtime/test_subproc.py [hunk 1]
```diff
@@ -55,10 +61,12 @@ def kernel_dot(Z):
         tl.store(Z + offs, z)
 
     src = ASTSource(fn=kernel_dot, signature={0: "*fp32"}, attrs=attrs, constants=dict())
-    triton.compile(src=src, target=("cuda", capability))
+    triton.compile(src=src, target=("xpu", capability))
 
 
 def test_compile_in_forked_subproc() -> None:
+    pytest.skip("FIXME: Port get_device_capability to XPU")
```
Comment (etiotto) [inline python/test/unit/runtime/test_subproc.py:68]: What is the issue to track this problem ?

File: python/test/unit/runtime/test_subproc.py [hunk 2]
```diff
@@ -30,10 +34,12 @@ def kernel_sub(a, b, o, N: tl.constexpr):
         signature={0: "*fp32", 1: "*fp32", 2: "*fp32"},
         attrs=attrs,
     )
-    triton.compile(src=src, target=("cuda", capability))
+    triton.compile(src=src, target=("xpu", capability))
 
 
 def test_compile_in_subproc() -> None:
+    pytest.skip("FIXME: Port get_device_capability to XPU")
```
Comment (etiotto) [inline python/test/unit/runtime/test_subproc.py:41]: What is the issue to track this problem ?

File: python/test/unit/runtime/test_cache.py [hunk 3]
```diff
@@ -257,3 +260,5 @@ def kernel(in_ptr0, out_ptr0, xnumel, XBLOCK: tl.constexpr):
         x0 = xindex
         tmp0 = tl.load(in_ptr0 + (x0), xmask)
         tl.store(out_ptr0 + (x0 + tl.zeros([XBLOCK], tl.int32)), tmp0, xmask)
+
+    reset_tmp_dir()
```
Comment (etiotto) [inline python/test/unit/runtime/test_cache.py:264]: why do we need this ?

File: python/test/unit/runtime/test_subproc.py (hunk 1 above)
Comment (whitneywhtsang) [reply python/test/unit/runtime/test_subproc.py:68]: https://github.com/intel/intel-xpu-backend-for-triton/issues/228

File: python/test/unit/runtime/test_subproc.py (hunk 2 above)
Comment (whitneywhtsang) [reply python/test/unit/runtime/test_subproc.py:41]: https://github.com/intel/intel-xpu-backend-for-triton/issues/228

File: python/test/unit/runtime/test_cache.py (hunk 3 above)
Comment (whitneywhtsang) [reply python/test/unit/runtime/test_cache.py:264]: yes, or else there is a leftover .tmp folder.

Comment (etiotto) [reply python/test/unit/runtime/test_cache.py:264]: For the record we have removed this change in a subsequent PR.

**Pull request merged**